Sources/Core/GTMSessionFetcher.m

```diff
@@ -1788,8 +1788,27 @@ - (void)endBackgroundTask {
 - (void)authorizeRequest {
   GTMSessionCheckNotSynchronized(self);
 
+  BOOL stopped = NO;
   @synchronized(self) {
-    _delayState = kDelayStateAuthorizing;
+    if (_userStoppedFetching) {
+      stopped = YES;
+    } else {
+      // Go into the delayed state for getting the authorization.
+      _delayState = kDelayStateAuthorizing;
+    }
+  }
+
+  if (stopped) {
+    // We end up here if someone calls `stopFetching` from another thread/queue while
+    // the fetch was being started up, so while `stopFetching` did the needed shutdown
+    // we have to ensure the requested callback was triggered.
+    if (self.stopFetchingTriggersCompletionHandler) {
+      NSError *error = [NSError errorWithDomain:kGTMSessionFetcherErrorDomain
+                                           code:GTMSessionFetcherErrorUserCancelled
+                                       userInfo:nil];
```

Comment:
Any useful info to put in the userInfo like the URL?

Reply:
Same as the other PR, will look at this across the board in the future.